Removed Requirement Pins

mrevalski opened the pull request:
The package versions listed in the requirements.txt file were pinned to fixed versions of each package. This ensured stability during initial development, but defeats the purpose of the weekly CI checks, which test for compatibility with the latest versions of packages. We have changed all package pins to ">=" in the requirements.txt file and updated the notebook version history.

mgough-970 commented:
The workflow fails to install dependencies, it appears that astrocut is set to an incompatable version in hstcal 

```
Using Python 3.11.12 environment at: /home/runner/micromamba/envs/hstcal
  × Failed to build `astrocut==0.6`
  ├─▶ The build backend returned an error
  ╰─▶ Call to
      `setuptools.build_meta:__legacy__.prepare_metadata_for_build_wheel`
      failed (exit status: 1)

      [stderr]
      Traceback (most recent call last):
        File "<string>", line 14, in <module>
        File
      "/home/runner/micromamba/envs/hstcal/lib/python3.11/site-packages/setuptools/build_meta.py",
      line 374, in prepare_metadata_for_build_wheel
          self.run_setup()
        File
      "/home/runner/micromamba/envs/hstcal/lib/python3.11/site-packages/setuptools/build_meta.py",
      line 512, in run_setup
          super().run_setup(setup_script=setup_script)
        File
      "/home/runner/micromamba/envs/hstcal/lib/python3.11/site-packages/setuptools/build_meta.py",
      line 317, in run_setup
          exec(code, locals())
        File "<string>", line 31, in <module>
        File
      "/home/runner/work/_temp/setup-uv-cache/sdists-v9/pypi/astrocut/0.6/E15ojwGnk2gnnpXKP5ltQ/src/ah_bootstrap.py",
      line 137, in <module>
          from setuptools.package_index import PackageIndex
      ModuleNotFoundError: No module named 'setuptools.package_index'

      hint: This usually indicates a problem with the package or the build
      environment.
  help: `astrocut` (v0.6) was included because `drizzlepac` (v3.9.1) depends
        on `astrocut<=0.9`
Error: Process completed with exit code 1.
```

mrevalski commented:
Hi @FDauphin, @cshanahan1, and @gibsongreen. This revised notebook is now ready for review! The details are below:

The package versions listed in the requirements.txt file were pinned to fixed versions of each package. This ensured stability during initial development, but defeats the purpose of the weekly CI checks, which test for compatibility with the latest versions of packages. I have changed all package pins to ">=" in the requirements.txt file and updated the notebook version history.

In addition to the changes to the requirements.txt file described above, several minor modifications have been made for compatibility with the latest photutils. Specifically, calls to `make_residual_image()` were updated so the `psf_shape` is now explicitly specified, rather than the previous implementation where it was an optional argument. In addition, a minor change was made in psf_utilities.py so that the matplotlib `vmin` cannot be less than zero when using `norm=LogNorm()`, as this creates an error message in the latest versions of matplotlib. Finally, a `clear_output()` command has been added when plotting the cutouts, or else the HTML rendering takes a long time to scroll through. A comment is included instructing users to remove this command if they would like to see all of the cutouts plotted in the notebook. Finally, I note that the update from photutils `v1.12.0` to `v2.2.0` results in some minor differences in photometry, typically < 0.5%, but as large as ~2% in some cases.

mrevalski commented:
Thank you for the swift approval @cshanahan1! If you also agree with these minor changes @FDauphin, I'll merge into main?
